src/ontodia/editor/editorController.tsx

```diff
@@ -513,6 +517,67 @@ export class EditorController {
         }
 
         this.setSelection(placedElements);
+
+        this.source.trigger('addElements', { elements: placedElements });
+    }
+
+    onAddElementsInConnectionMenu(
+        elementIris: ElementIri[],
+        targetElement: Element,
+        linkType: FatLinkType|undefined,
+    ) {
+        const batch = this.view.model.history.startBatch();
+        const targetBounds = boundsOf(targetElement);
+        const targetPosition: Vector = {
+            x: targetBounds.x + targetBounds.width / 2,
+            y: targetBounds.y + targetBounds.height / 2,
+        };
+        let startAngle: number;
+        if (targetElement.links.length > 0) {
+            let xSum = 0;
+            let ySum = 0;
+            for (const link of targetElement.links) {
+                const linkSource = this.model.sourceOf(link);
+                const source = linkSource !== targetElement ? linkSource : this.model.targetOf(link);
+
+                xSum += source.position.x + source.size.width / 2;
+                ySum += source.position.y + source.size.height / 2;
+            }
+            const averageSourcePosition: Vector = {
+                x: xSum / targetElement.links.length,
+                y: ySum / targetElement.links.length,
+            };
+            const vectorDiff: Vector = {
+                x: targetPosition.x - averageSourcePosition.x,
+                y: targetPosition.y - averageSourcePosition.y,
+            };
+            if (vectorDiff.x !== 0 || vectorDiff.y !== 0) {
+                startAngle = Math.atan2(vectorDiff.y, vectorDiff.x);
+            }
+        }
+        const placedElements = placeElementsAround({
+            view: this.view,
+            elementIris,
+            zeroPosition: targetPosition,
+            startAngle,
+            callBack: (elements) => {
```

Comment:
We should fire event only after operation finish, when elements have positions and sizes.